File: backend/prompt/report.py
REPORT_ANALYSIS_PROMPT = """
你是 Synthetic User Lab 的运行报告分析器。

你会收到一份结构化的事实摘要（非原始步骤日志），请基于这些事实生成面向产品经理和设计师的运行报告。

## 你的输出

必须输出一个 JSON 对象，字段只能是：
- summary: 一段 2-4 句的中文总结，描述本次 run 的整体结果、关键问题和影响用户
- conclusion: 只能是 keep / optimize / fix
  - keep: 本次 run 没有明显问题
  - optimize: 基本逻辑正确但存在体验优化空间
  - fix: 存在明显问题需要修复
- key_findings: 中文字符串数组，基于事实的具体发现（不要写"共执行N步"这类统计废话）
- next_recommendations: 中文字符串数组，针对具体 UI 元素和操作路径的修复/优化建议

## 约束

- 必须基于输入的事实摘要推理，不能编造不存在的证据
- key_findings 每条必须指向具体步骤或具体 UI 问题
- next_recommendations 最多 10 条；无问题则输出 []
- 没有明显问题时，不要强行制造问题，summary 简洁确认成功即可
- 绝对禁止输出与本次 run 无关的开发规划建议
""".strip()

File: frontend/src/types/report.ts
import type { Persona } from './persona';
import type { Task, ActionName } from './task';
import type { RunStatus, RunErrorType } from './run';

export type ReportConclusion = 'keep' | 'optimize' | 'fix';
export type FrictionSeverity = 'low' | 'medium' | 'high';
export type ValidationStatus = 'running' | 'succeeded' | 'failed';
export type WaitObservationStatus = 'success' | 'actionable' | 'normal_timeout' | 'abnormal_stuck';
export type WaitObservationDecision = 'normal_waiting' | 'abnormal_stuck' | 'ready_for_next_action' | 'task_completed';

// ============================ Report ============================ #

export interface KeyScreenshot {
  label: string;
  step_index: number;
  path: string;
  source: 'before_action' | 'after_action' | 'wait_observation';
}

export interface FrictionIssue {
  signal: string;
  severity: FrictionSeverity;
  step_indexes: number[];
  description: string;
  suggested_fix: string;
}

export interface RunReport {
  run_id: string;
  status: RunStatus;
  summary: string;
  success: boolean;
  conclusion: ReportConclusion;
  persona: Persona;
  task: Task;
  total_steps: number;
  friction_signals: string[];
  friction_issues: FrictionIssue[];
  key_findings: string[];
  next_recommendations: string[];
  step_details: Record<string, unknown>[];
  structured_facts: Record<string, unknown> | null;
  key_screenshots: KeyScreenshot[];
  error_type?: RunErrorType | null;
  error_message?: string | null;
}

// ============================ Step ============================ #

export interface ObservedElement {
  text: string;
  selector: string;
}

export interface FormFieldState {
  name: string;
  selector: string;
  value: string;
}

export interface ObservedPageState {
  current_url: string;
  title: string;
  visible_text_summary: string;
  clickable_elements: ObservedElement[];
  form_fields: FormFieldState[];
  error_messages: string[];
  screenshot_path: string | null;
}

export interface ActionInput {
  action: ActionName;
  payload: Record<string, unknown>;
  reason: string;
}

export interface ExecutionResult {
  action: ActionName;
  success: boolean;
  detail: string;
  error_message?: string | null;
}

export interface ValidationResult {
  status: ValidationStatus;
  should_stop: boolean;
  progress_summary: string;
  friction_signals: string[];
  detected_success: boolean;
  detected_error: boolean;
}

export interface RetrievedContextItem {
  source_type: string;
  title: string;
  content: string;
  source_ref: string;
}

export interface StepLog {
  step_index: number;
  observed_page_state: ObservedPageState;
  decided_action: ActionInput;
  execution_result: ExecutionResult;
  validation_result: ValidationResult;
  post_action_page_state: ObservedPageState;
  wait_observation_status?: WaitObservationStatus | null;
  wait_observation_reason?: string | null;
  wait_observation_observations?: number | null;
  wait_observation_elapsed_ms?: number | null;
  wait_observation_timeout_ms?: number | null;
  wait_observation_terminal_decision?: WaitObservationDecision | null;
  wait_observation_traces?: Record<string, unknown>[];
  retrieval_context?: RetrievedContextItem[];
  before_page_state: ObservedPageState;
  after_page_state: ObservedPageState;
}
